lib/cretonne/meta/semantics/types.py

```diff
@@ -0,0 +1,9 @@
+"""
+The semantics.types module predefines all the Cretone primitive bitvector
+types.
+"""
+from cdsl.types import BVType
+from cdsl.typevar import MAX_BITVEC, int_log2
+
+for width in range(0, int_log2(MAX_BITVEC)+1):
+    BVType(2**width)
```

Comment:
The `BVType.with_bits()` is a bit gross, looking up existing types by name.

How about adding a `BVType._registry` dictionary that maps bits to the corresponding bitvector type. Then `with_bits` can generate types on demand and add then to the registry. This is OK for the bitvectors since they don't appear in rust code.